numpy/random/mtrand/mtrand.pyx

```diff
@@ -1426,15 +1435,272 @@ cdef class RandomState:
         >>> plt.show()
 
         """
+        if sizeof(long) == sizeof(npy_int64):
+            return self.random_int64(low, high, size)
+        elif sizeof(long) == sizeof(npy_int32):
+            return self.random_int32(low, high, size)
+        else:
+            msg = 'Unimplemented for sizeof(long) == %d' % sizeof(long)
+            raise RuntimeError(msg)
+
+    def random_intp(self, low, high=None, size=None):
```

Comment:
Not really. They have different uses, `np.intp` is the indexing integer, and we need to keep `np.int` as is for backwards compatibility. Python 2 had two integer types, `int`, and `long`, not the same as the C types of the same name. The `int` type was implemented as C `long`, which was efficient, and the later `long` type has arbitrary precision. Python 3 has only the `long` type, but we need to carry along the historic crud so as to not break old code/results.